src/statd/statd.c

```diff
@@ -566,6 +572,12 @@ int main(int argc, char *argv[])
 		sr_disconnect(sr_conn);
 		return EXIT_FAILURE;
 	}
+	err = sub_to_hardware(&statd);
```

Comment:
I think it's time to make all these subscriptions calls into a function. Something like `int subscribe(&statd)` which does all these `sub_to` calls without the `sr_disconnect()` in each.

Reply:
Yes, it is a good thing for the future to do that.